### ISSUE TITLE
configure: error: The powerpc64le is not supported.

### ISSUE THREAD

**Issue by edowson** [crosstool-ng/crosstool-ng#1227]
I'm tried using crosstool-ng to build a toolchain for an IBM Power9 system on an Ubuntu-18.04 x86_64 host. The target triple that I want is ppc64le-unknown-linux-gnu.

When I try to build the toolchain, I get the following error:

```bash
[INFO ]    Building for multilib 1/1: ''
[ERROR]      checking sysdep dirs... configure: error: The powerpc64le is not supported.
[ERROR]   
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Building for multilib 1/1: '''
[ERROR]  >>        called in step 'Installing C library headers & start files'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@376]
[ERROR]  >>        called from: glibc_backend_once[scripts/build/libc/glibc.sh@263]
[ERROR]  >>        called from: CT_IterateMultilibs[scripts/functions@1586]
[ERROR]  >>        called from: glibc_backend[scripts/build/libc/glibc.sh@74]
[ERROR]  >>        called from: glibc_start_files[scripts/build/libc/glibc.sh@38]
[ERROR]  >>        called from: do_libc_start_files[scripts/build/libc.sh@28]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@696]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >> NOTE: You configuration uses non-default patch sets. Please
[ERROR]  >> select 'bundled' as the set of patches applied and attempt
[ERROR]  >> to reproduce this issue. Issues reported with other patch
[ERROR]  >> set selections (none, local, bundled+local) are going to be
[ERROR]  >> closed without explanation.
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]   
[ERROR]  (elapsed: 7:45.59)
[07:46] / /mnt/storage/mount/project/software/tool/crosstool-ng/ct-ng:261: recipe for target 'build' failed
make[1]: *** [build] Error 1
```

- crosstool-NG version
crosstool-NG version 1.24.0.6-afaf7b9

- Host OS and its version
Ubuntu-18.04, x86_64 linux

- Attach `.config`
[ppc64le-unknown-linux-gnu.linux.config.txt](https://github.com/crosstool-ng/crosstool-ng/files/3565405/ppc64le-unknown-linux-gnu.linux.config.txt)

- Attach `build.log`
[build.log](https://github.com/crosstool-ng/crosstool-ng/files/3565410/build.log)

**Comment by edowson** [comment]
I tried building once again, using the existing powerpc64le sample, but still get a build error:

```bash
mkdir build-powerpc64le-unknown-linux-gnu; cd build-powerpc64le-unknown-linux-gnu;
../crosstool-ng/ct-ng powerpc64le-unknown-linux-gnu
../crosstool-ng/ct-ng menuconfig
unset LD_LIBRARY_PATH
../crosstool-ng/ct-ng build.`nproc`
```

```bash
[INFO ]  Installing pass-2 core C gcc compiler: done in 173.07s (at 15:57)
[INFO ]  =================================================================
[INFO ]  Installing C library
[INFO ]    =================================================================
[INFO ]    Building for multilib 1/1: ''
[EXTRA]      Cleaning up start files
[EXTRA]      Configuring C library
[EXTRA]      Building C library
[ERROR]      collect2: error: ld returned 1 exit status
[ERROR]      make[4]: *** [/mnt/storage/mount/project/software/tool/build-powerpc/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc.so] Error 1
[ERROR]      make[3]: *** [elf/subdir_lib] Error 2
[ERROR]      make[2]: *** [all] Error 2
[ERROR]   
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Building for multilib 1/1: '''
[ERROR]  >>        called in step 'Installing C library'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@376]
[ERROR]  >>        called from: glibc_backend_once[scripts/build/libc/glibc.sh@395]
[ERROR]  >>        called from: CT_IterateMultilibs[scripts/functions@1586]
[ERROR]  >>        called from: glibc_backend[scripts/build/libc/glibc.sh@74]
[ERROR]  >>        called from: glibc_main[scripts/build/libc/glibc.sh@44]
[ERROR]  >>        called from: do_libc_main[scripts/build/libc.sh@33]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@696]
[ERROR]  >>
```

- Attach `.config`:
[build-powerpc64le.config.txt](https://github.com/crosstool-ng/crosstool-ng/files/3565577/build-powerpc64le.config.txt)

The build.log file for this particular build is too large (34MB) to attach here, but it is the default config, so it can be reproduced.

**Comment by edowson** [comment]
I just tried building `powerpc64le-unknown-linux-gnu` directly on an IBM Power9 system, and I get the following build error:

```
[INFO ]  =================================================================
[INFO ]  Installing C library
[INFO ]    =================================================================
[INFO ]    Building for multilib 1/1: ''
[EXTRA]      Cleaning up start files
[EXTRA]      Configuring C library
[EXTRA]      Building C library
[ERROR]      collect2: error: ld returned 1 exit status
[ERROR]      make[4]: *** [/project/software/tool/build-powerpc64le-unknown-linux-gnu/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc.so] Error 1
[ERROR]      make[3]: *** [elf/subdir_lib] Error 2
[ERROR]      make[2]: *** [all] Error 2
[ERROR]   
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Building for multilib 1/1: '''
[ERROR]  >>        called in step 'Installing C library'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@376]
[ERROR]  >>        called from: glibc_backend_once[scripts/build/libc/glibc.sh@395]
[ERROR]  >>        called from: CT_IterateMultilibs[scripts/functions@1586]
[ERROR]  >>        called from: glibc_backend[scripts/build/libc/glibc.sh@74]
[ERROR]  >>        called from: glibc_main[scripts/build/libc/glibc.sh@44]
[ERROR]  >>        called from: do_libc_main[scripts/build/libc.sh@33]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@696]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]   
[ERROR]  (elapsed: 7:22.37)
[07:22] / /project/software/tool/crosstool-ng/ct-ng:261: recipe for target 'build' failed
make[1]: *** [build] Error 2
../crosstool-ng/ct-ng:265: recipe for target 'build.96' failed
make: *** [build.96] Error 2
```

- Attach `.config`
[powerpc64le.config.txt](https://github.com/crosstool-ng/crosstool-ng/files/3566752/powerpc64le.config.txt)

- Attach `build.log`

The last few lines leading up to the error in the `build.log` file are as follows:
```bash
[ERROR]      collect2: error: ld returned 1 exit status
[ALL  ]      ../Makerules:699: recipe for target '/project/software/tool/build-powerpc64le-unknown-linux-gnu/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc.so' failed
[ERROR]      make[4]: *** [/project/software/tool/build-powerpc64le-unknown-linux-gnu/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc.so] Error 1
[ALL  ]      make[4]: Leaving directory '/project/software/tool/build-powerpc64le-unknown-linux-gnu/.build/src/glibc-2.29/elf'
[ALL  ]      Makefile:258: recipe for target 'elf/subdir_lib' failed
[ERROR]      make[3]: *** [elf/subdir_lib] Error 2
[ALL  ]      make[3]: Leaving directory '/project/software/tool/build-powerpc64le-unknown-linux-gnu/.build/src/glibc-2.29'
[ALL  ]      Makefile:9: recipe for target 'all' failed
[ERROR]      make[2]: *** [all] Error 2
[ALL  ]      make[2]: Leaving directory '/project/software/tool/build-powerpc64le-unknown-linux-gnu/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib'
[ERROR]  
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Building for multilib 1/1: '''
[ERROR]  >>        called in step 'Installing C library'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@376]
[ERROR]  >>        called from: glibc_backend_once[scripts/build/libc/glibc.sh@395]
[ERROR]  >>        called from: CT_IterateMultilibs[scripts/functions@1586]
[ERROR]  >>        called from: glibc_backend[scripts/build/libc/glibc.sh@74]
[ERROR]  >>        called from: glibc_main[scripts/build/libc/glibc.sh@44]
[ERROR]  >>        called from: do_libc_main[scripts/build/libc.sh@33]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@696]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]  
[ERROR]  (elapsed: 7:22.37)
```

**Comment by cpackham** [comment]
The first error seems to be due to an old libc (CT_GLIBC_V_2_17=y) being selected which doesn't seem to like powerpc64le (I'm not going to dig into that one any further).

The other errors it's hard to tell because you haven't attached a build.log from any of them. There are a few open issues which all seem to relate to multilib support so maybe that's the common issue.

I'm attempting to reproduce your issue with the powerpc64le-unknown-linux-gnu sample. I'll report back on that.

**Comment by cpackham** [comment]
So this worked for me in the following environment

```
$ cat >Dockerfile <<EOF
FROM debian:buster-slim

ENV DEBIAN_FRONTEND="noninteractive"

RUN apt-get update
RUN apt-get install -y \
        autoconf \
        automake \
        autotools-dev \
        bison \
        build-essential \
        bzip2 \
        curl \
        flex \
        gawk \
        git \
        gperf \
        help2man \
        libncurses5-dev \
        libncursesw5-dev \
        libtool-bin \
        subversion \
        texinfo \
        unzip \
        wget

WORKDIR /usr/src
RUN git clone https://github.com/crosstool-ng/crosstool-ng.git

WORKDIR /usr/src/crosstool-ng
RUN ./bootstrap && \
        ./configure --prefix=/usr && \
        make && make install

RUN useradd -ms /bin/bash x-tool

USER x-tool
WORKDIR /home/x-tool

RUN ct-ng powerpc64le-unknown-linux-gnu
RUN ct-ng build

ENTRYPOINT ["/bin/bash"]
EOF

$ docker build . -t test
$ docker run --rm -it test
```

One thing about this environment is there is no python3 (or python) so glibc-2.29 is excluded automatically

**Comment by cpackham** [comment]
Right adding python3 to my environment and thus enabling glibc-2.29 I can reproduce the problem. Here's the snippet from build.log (as soon as I can figure out how to attach files I'll add the whole thing.

```
/home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/../../../../powerpc64le-unknown-linux-gnu/bin/ld: /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/libgcc.a(ppc64-fp.o): in function `__fixdfdi':
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/src/gcc/libgcc/config/rs6000/ppc64-fp.c:57: multiple definition of `__fixdfdi'; /home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc_pic.os:/home/x-tool/.build/src/glibc-2.29/soft-fp/fixdfdi.c:36: first defined here
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/../../../../powerpc64le-unknown-linux-gnu/bin/ld: /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/libgcc.a(ppc64-fp.o): in function `__fixsfdi':
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/src/gcc/libgcc/config/rs6000/ppc64-fp.c:65: multiple definition of `__fixsfdi'; /home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc_pic.os:/home/x-tool/.build/src/glibc-2.29/soft-fp/fixsfdi.c:36: first defined here
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/../../../../powerpc64le-unknown-linux-gnu/bin/ld: /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/libgcc.a(ppc64-fp.o): in function `__fixunsdfsi':
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/src/gcc/libgcc/config/rs6000/ppc64-fp.c:73: multiple definition of `__fixunsdfsi'; /home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc_pic.os:/home/x-tool/.build/src/glibc-2.29/soft-fp/fixunsdfsi.c:36: first defined here
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/../../../../powerpc64le-unknown-linux-gnu/bin/ld: /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/libgcc.a(ppc64-fp.o): in function `__fixunssfsi':
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/src/gcc/libgcc/config/rs6000/ppc64-fp.c:82: multiple definition of `__fixunssfsi'; /home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc_pic.os:/home/x-tool/.build/src/glibc-2.29/soft-fp/fixunssfsi.c:36: first defined here
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/../../../../powerpc64le-unknown-linux-gnu/bin/ld: /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/libgcc.a(ppc64-fp.o): in function `__floatdidf':
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/src/gcc/libgcc/config/rs6000/ppc64-fp.c:115: multiple definition of `__floatdidf'; /home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc_pic.os:/home/x-tool/.build/src/glibc-2.29/soft-fp/floatdidf.c:36: first defined here
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/../../../../powerpc64le-unknown-linux-gnu/bin/ld: /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/libgcc.a(ppc64-fp.o): in function `__floatundidf':
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/src/gcc/libgcc/config/rs6000/ppc64-fp.c:127: multiple definition of `__floatundidf'; /home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc_pic.os:/home/x-tool/.build/src/glibc-2.29/soft-fp/floatundidf.c:36: first defined here
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/../../../../powerpc64le-unknown-linux-gnu/bin/ld: /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/libgcc.a(ppc64-fp.o): in function `__floatdisf':
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/src/gcc/libgcc/config/rs6000/ppc64-fp.c:139: multiple definition of `__floatdisf'; /home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc_pic.os:/home/x-tool/.build/src/glibc-2.29/soft-fp/floatdisf.c:36: first defined here
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/../../../../powerpc64le-unknown-linux-gnu/bin/ld: /home/x-tool/.build/powerpc64le-unknown-linux-gnu/buildtools/lib/gcc/powerpc64le-unknown-linux-gnu/8.3.0/libgcc.a(ppc64-fp.o): in function `__floatundisf':
[ALL  ]      /home/x-tool/.build/powerpc64le-unknown-linux-gnu/src/gcc/libgcc/config/rs6000/ppc64-fp.c:164: multiple definition of `__floatundisf'; /home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc_pic.os:/home/x-tool/.build/src/glibc-2.29/soft-fp/floatundisf.c:36: first defined here
[ERROR]      collect2: error: ld returned 1 exit status
[ERROR]      make[3]: *** [../Makerules:699: /home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib/libc.so] Error 1
[ALL  ]      make[3]: Leaving directory '/home/x-tool/.build/src/glibc-2.29/elf'
[ERROR]      make[2]: *** [Makefile:258: elf/subdir_lib] Error 2
[ALL  ]      make[2]: Leaving directory '/home/x-tool/.build/src/glibc-2.29'
[ERROR]      make[1]: *** [Makefile:9: all] Error 2
[ALL  ]      make[1]: Leaving directory '/home/x-tool/.build/powerpc64le-unknown-linux-gnu/build/build-libc-final/multilib
```

I'm tempted to say this is a regression in glibc but maybe there's some new option crosstool-ng needs to be passing. @edowson as a workaround is downgrading to glibc 2.28 and option for you?

**Comment by cpackham** [comment]
looks like some how both the rs6000 and soft floating point support has been enabled.

**Comment by edowson** [comment]
@cpackham I'm okay with downgrading to glibc-2.28.

On the x86_64 system, `ldd` shows that I have glibc-2.27 installed. So, any version of glibc-2.27 or higher would be fine.

Would you be able to fix the issue with rs6000 and soft-floating point support being enabled. I had explicitly set FPU to hard in the `.config` file.

**Comment by cpackham** [comment]
I've reported the bug upstream https://sourceware.org/bugzilla/show_bug.cgi?id=24966.

**Comment by cpackham** [comment]
Looks like glibc-2.30 doesn't have this issue. Just skimming the changes in the glibc git repo I can't spot anything obvious that addresses the issue. I have got a PR #1232 that adds the new glibc version you could take that for a spin.

**Comment by cpackham** [comment]
I think I know what the problem is.

https://github.com/crosstool-ng/crosstool-ng/blob/afaf7b9a25b5d77991002936be5c47fc5ff549de/packages/glibc/2.29/0001-Add-ARC-architecture.patch#L2296

The ARC patch ends up setting with_fp_cond for non-arc systems. Which turns off floating point support (which is supposed to be mandatory for powerpc64). Deleting packages/glibc/2.29/0001-Add-ARC-architecture.patch would fix the build issue for you. There may be a less drastic fix (i.e. modify the patch).

**Comment by edowson** [comment]
@cpackham I managed to get the toolchain built on a Power9 system. I've installed the built toolchain into a separate folder. Why do I get a `GLIBC_2.29 not found` error, when I try to run `./clang --version` from my custom sysroot location?

```bash
./clang --version
./clang: /lib/powerpc64le-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /project/software/library/unrealengine-4.24-ppc64le-master/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_ppc64le/v14_clang-7.0.1-centos7/powerpc64le-unknown-linux-gnu/bin/../lib/libstdc++.so.6)
```

**Comment by cpackham** [comment]
@edowson I'm not sure how clang is going to work with a glibc compiled for gcc. I guess it might work if you can tell it the right sysroot (e.g --sysroot=/path/to/sysroot) although I'm used to the compiler being tightly coupled to the libc.

Are you able to just use the generated gcc instead of clang? (based on the paths I'm guessing you need clang for some kind of custom language front-end).

Another option that I have used for Rust is to convince llvm (i.e. clang) to output assembly and use the gcc assembler/linker.

**Comment by edowson** [comment]
@cpackham I rebuilt the powerpc64le toolchain using `glibc-2.27` instead of the default `glibc-2.29`, and it picks up the system installed `glibc-2.27`.

**Comment by edowson** [comment]
@cpackham I think if I set LD_LIBRARY_PATH as follows, it might resolve the issue:

```bash
unset LD_LIBRARY_PATH
local LD_LIBRARY_PATH="$LIBCXX_LIB_PATH:$CLANG_TOOLCHAIN_SYSROOT_PATH:/lib64;$CLANG_TOOLCHAIN_SYSROOT_PATH/usr/lib64:/usr/local/cuda/lib64:$SYSTEM_LIB_PATH"
```

**Comment by qhaas** [comment]
> The first error seems to be due to an old libc (CT_GLIBC_V_2_17=y) being selected which doesn't seem to like powerpc64le (I'm not going to dig into that one any further).

GLIBC 2.17 on ppc64le occupies an extant niche due to supercomputers like [Sierra](https://www.top500.org/system/179398) and [Summit](https://www.top500.org/system/179397) running ppc64le Red Hat Enterprise Linux 7 (RHEL7).

I adjusted the default 'powerpc64le-unknown-linux-gnu' [config target](https://github.com/crosstool-ng/crosstool-ng/files/4376855/1_24_0.txt) to use the same major/minor version of the kernel (4.14), glibc (2.17), and binutils (2.27) of ppc64le RHEL7 and encountered this same error during the building of the toolchain 'error: The powerpc64le is not supported'.



It appeared in both latest stable (b2151f1dba2b20c310adfe7198e461ec4469172b, crosstool-ng-1.24.0) and master (be1b56676828c80045bff401385a75ec0f480145) on a power9 system in RHEL7.

In [another discussion](https://stackoverflow.com/a/45141873), they mention that upstream ppc64le landed in glibc 2.19, so RH must have backported ppc64le support in their branch of glibc 2.17, unfortunately, I don't see it being easy to apply all of what they've done to the flavor used by crosstool-ng...

For those lacking a ppc64le system and wanting to give it a shot, one can use the [attached dockerfile](https://github.com/crosstool-ng/crosstool-ng/files/4376842/ppc64le_on_x8664.txt) on x86-64 systems by leveraging [qemu](https://ownyourbits.com/2018/06/27/running-and-building-arm-docker-containers-in-x86/).  In x86-64 Ubuntu 18.04, deploy packages `qemu-user qemu-user-static binutils-powerpc64le-linux-gnu`.

**Comment by edowson** [comment]
Just delete the patch file that gets downloaded when building for ppc64le.
